### PR TITLE
Adding Markduplicates from picard 2.8.1

### PR DESCRIPTION
**Pull request by rhshah**
:sparkles: :sparkles: :sparkles:

:ballot_box_with_check: Adding docker file for 2.81
:ballot_box_with_check:  Adding CWL for running MarkDuplicates

#11

### CODE REVIEW

File: picard_mark_duplicates_2.8.1/container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    R version ${R_VERSION}"
+
+# Install ant, git for building
+RUN apt-get update && \
+    apt-get --no-install-recommends install -y \
+    git \
+    unzip \
+    wget \
+    r-base \
+    ant && \
```
Comment (ionox0) [inline picard_mark_duplicates_2.8.1/container/Dockerfile]: is ant required to build picard?

File: picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      prefix: I=
+      separate: false
+    doc: Input file (bam or sam).  Required.
+  - id: output
+    type: string
+    inputBinding:
+      position: 0
+      prefix: O=
+      separate: false
+      valueFrom: '$(inputs.input.basename.replace(/.bam/, ''_md.bam''))'
```
Comment (ionox0) [inline picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl]: I'm a bit worried that these double quotes aren't supposed to be there, how does it show up in the expression editor in rabix?

File: picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    outputBinding:
+      glob: '$(inputs.input.basename.replace(/.bam/, ''_md.bam''))'
+    secondaryFiles:
+      - ^.bai
+      - ^.metrics
+label: picard_mark_duplicates_2.8.1
+arguments:
+  - position: 0
+    prefix: ''
+    valueFrom: "${\n  if(inputs.memory_per_job && inputs.memory_overhead) {\n   \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if (inputs.memory_per_job && !inputs.memory_overhead){\n    \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if(!inputs.memory_per_job && inputs.memory_overhead){\n    \n    return \"-Xmx15G\"\n  }\n  else {\n    \n  \treturn \"-Xmx15G\"\n  }\n}"
```
Comment (ionox0) [inline picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl]: this is a bit hard to review in a PR but maybe eventually we can think of a way to get this to print more nicely in the CWL

File: picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        'foaf:mbox': 'mailto:shahr2@mskcc.org'
+        'foaf:name': Ronak Shah
+    'foaf:name': Memorial Sloan Kettering Cancer Center
+'doap:release':
+  - class: 'doap:Version'
+    'doap:name': picard
+    'doap:revision': 2.8.1
+  - class: 'doap:Version'
+    'doap:name': cwl-wrapper
+    'doap:revision': 1.0.0
```
Comment (ionox0) [inline picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl]: would be nice to get rabix to put this at the top of the CWL but it's a minor thing I think

File: picard_mark_duplicates_2.8.1/container/Dockerfile (hunk 1 above)
Comment (rhshah) [reply picard_mark_duplicates_2.8.1/container/Dockerfile]: ♻️

File: picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl (hunk 2 above)
Comment (rhshah) [reply picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl]: Works with cwltool and Rabix.

File: picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl (hunk 3 above)
Comment (rhshah) [reply picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl]: bitmoji

File: picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl (hunk 4 above)
Comment (rhshah) [reply picard_mark_duplicates_2.8.1/picard_mark_duplicates_2.8.1.cwl]: bitmoji

**Pull request merged**